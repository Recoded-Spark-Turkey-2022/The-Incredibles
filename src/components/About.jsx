import React from 'react';
import AboutHero from '../assets/pics/about-hero.svg';
import AboutStar from '../assets/pics/about-star.svg';

function About() {
  return (
    <section name="about" className="bg-emerald-50">
      <div name="about-header" className="text-center p-20 mx-10 ...">
        <h1 className="text-cyan-600 font-bold text-4xl">About</h1>
        <br />
        <p className="mx-20">
          Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nulla est
          non repellat officia odio. Ipsam veniam, laudantium dolores
          praesentium nisi soluta impedit quae, suscipit fugit eveniet, qui
          tempora fugiat veritatis! Lorem ipsum dolor sit amet consectetur
          adipisicing elit. Vitae nostrum, sint eius voluptatum cum ratione
          labore expedita similique iste facilis sed totam cupiditate vero
          inventore reiciendis dolorem dolorum accusantium voluptatibus!
        </p>
        <br />
        <button
          type="button"
          className="inline-block px-10 py-2.5 bg-cyan-600 text-white font-medium text-l leading-tight 
        rounded-full shadow-md
        ease-in duration-300 hover:bg-purple-700 hover:shadow-lg my-9 hover:scale-110"
        >
          Learn More
        </button>
      </div>
      <div name="about-body" className="flex justify-between">
        <div name="cards" className="flex justify-start">
          <div name="left-side-cards">
            <div className="p-6 max-w-sm mx-auto space-x-4">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Reliability
              </h2>
              <br />
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque
                cum debitis repellendus placeat maiores eaque corrupti rerum?
                Reiciendis nisi voluptas corrupti, aspernatur autem est vero
                necessitatibus optio laudantium rerum numquam.{' '}
              </p>
            </div>
            <div className="p-6 max-w-sm mx-auto space-x-4">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Efficiency
              </h2>
              <br />
              <p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit.
                Laboriosam, quidem sit minus dolore explicabo velit quas
                temporibus sequi vitae aut ducimus nostrum beatae eveniet atque
                ullam, quisquam aliquam sed est.
              </p>
            </div>
          </div>
          <div name="right-side-cards">
            <div className="p-6 max-w-sm mx-auto space-x-4">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Motivation
              </h2>
              <br />
              <p>
                Lorem ipsum dolor, sit amet consectetur adipisicing elit.
                Voluptates, quaerat, molestiae vitae sed aliquid in nostrum
                laborum obcaecati optio expedita atque ab ipsam veritatis iure
                iusto similique quibusdam laboriosam sunt.
              </p>
            </div>
            <div className="p-6 max-w-sm mx-auto space-x-4">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Creativity
              </h2>
              <br />
              <p>
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                Suscipit ex et commodi consequuntur, vel quis alias perferendis
                magni minima labore. Totam doloribus incidunt velit odio
                corrupti beatae quasi minima laudantium.
              </p>
            </div>
          </div>
        </div>
        <div name="side-hero" className="flex justify-end">
          <img src={AboutHero} alt="about-hero" />
        </div>
      </div>

      <div className="carousel-indicators right-0 bottom-0 left-0 flex justify-start pb-20 m-10">
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="0"
          className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-cyan-600 bg-cyan-600"
          aria-current="true"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="1"
          className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="2"
          className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="3"
          className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
      </div>
    </section>
  );
}

export default About;
