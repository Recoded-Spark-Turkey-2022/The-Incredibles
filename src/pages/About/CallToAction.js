import React from 'react';
import callToActionImg from '../../assets/pics/aboutpage/aboutpgCallToAction.svg';

function CallToAction() {
  return (
    <div className='container mx-auto text-center'>
      <div className=''>
        <img src={callToActionImg} className=' m-auto'/>
      </div>
      <div className=''>
        <h4 className='font-normal text-2xl text-cyan-600'>
        Share your story, help others and get help from others
        </h4>
        <div className='text-center w-2/4 m-auto pb-3 pt-5 '>
        <p className='text-zinc-400	font-light  '>
        But Brooke Chaffin and Catherine Connors are looking to change that with the introduction of Maverick, a social network that connects young girls with female mentors to express their creativity in a safe space.
        </p>
        </div>
        
      </div>
      <div>
      <button
          type="button"
          className=" h-10 px-10 py-2.5 bg-cyan-600 text-white font-medium text-l leading-tight 
             rounded-full shadow-md
             ease-in duration-300 hover:bg-purple-700 hover:shadow-lg  hover:scale-110"
        >
          Sign Up
        </button>
      </div>

    </div>
  );
}

export default CallToAction;
