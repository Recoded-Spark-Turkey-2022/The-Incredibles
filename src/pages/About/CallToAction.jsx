import React from 'react';
import Button from '../../components/Button';
import callToActionImg from '../../assets/pics/aboutpage/aboutpgCallToAction.svg';
import { useAuthState } from 'react-firebase-hooks/auth';
import { auth } from '../../firebase/firebase';

function CallToAction() {
  const [user] = useAuthState(auth);
  return (
    <div className="container mx-auto text-center">
      <div className="">
        <img src={callToActionImg} className=" m-auto" />
      </div>
      <div className="">
        <h4 className="font-normal text-2xl text-cyan-600">
          Share your story, help others and get help from others
        </h4>
        <div className="text-center w-2/4 m-auto pb-3 pt-5 ">
          <p className="text-zinc-400	font-light  ">
            After years of struggling as a refugee, while I was seeking on
            internet I found Refubook website and it changed my life immediately
            finally found a job that allowed me to provide for my family. I
            worked hard and saved every penny, determined to build a better
            future for my children. With the help of a kind Refubook community,
            I was able to put my kids through school and give them opportunities
            I never had. As I watch them pursue their dreams, I am filled with
            pride and gratitude for the journey that brought us here with this
            amazing Refubook members.
          </p>
        </div>
      </div>
      <div className={user ? 'hidden' : ''}>
        <Button name="Sign Up" path="/signup" />
      </div>
    </div>
  );
}

export default CallToAction;
