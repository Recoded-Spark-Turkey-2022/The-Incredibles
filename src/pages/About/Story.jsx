import React from 'react';
import AboutPHero from '../../assets/pics/aboutpage/aboutp-hero.svg';

function Story() {
  return (
    <section name="story" className="flex flex-col h-screen">
      <div
        name="hero-image"
        className="bg-gradient-to-t from-gray-300 to-transparent max-w-screen h-1/2"
      >
        <img src={AboutPHero} className="w-screen h-full" />
      </div>
      <div name="story" className="max-w-screen h-1/2">
        <h2 className="text-center text-6xl pt-20 pb-10 font-bold text-cyan-700">
          Our Story
        </h2>
        <div className="px-20 mx-20 pb-20">
          <p className="text-center px-20 mx-20 font text-xl max-w-screen">
            It is a long established fact that a reader will be distracted by
            the readable content of a page when looking at its layout. The point
            of using Lorem Ipsum is that it has a more-or-less normal
            distribution of letters, as opposed to using `Content here, content
            here`, making it look like readable English.
          </p>
        </div>
      </div>
    </section>
  );
}

export default Story;
