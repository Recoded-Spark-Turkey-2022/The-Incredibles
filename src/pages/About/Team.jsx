import React, { useState } from "react";

const MEMBERS=[
    {
        name: "Miray",
        occupation: "Jr Front-end Web Developer",
        country: "Turkey",
        id:0,
    },
    {
        name: "Rabia",
        occupation: "Jr Front-end Web Developer",
        country: "Turkey",
        id:1,
    },
    {
        name: "Batoul",
        occupation: "Jr Front-end Web Developer",
        country: "Turkey",
        id:2,
    },
    {
        name: "Ahmad",
        occupation: "Jr Front-end Web Developer",
        country: "Turkey",
        id:3,
    },
    {
        name: "Omer",
        occupation: "Jr Front-end Web Developer",
        country: "Turkey",
        id:4,
    },
    {
        name: "Osman",
        occupation: "Jr Front-end Web Developer",
        country: "USA",
        id:5,
    },
    {
        name: "Veli",
        occupation: "Jr Front-end Web Developer",
        country: "USA",
        id:6,
    },
    {
        name: "Ali",
        occupation: "Jr Front-end Web Developer",
        country: "Lebanon",
        id:7,
    },
    {
        name: "Osman",
        occupation: "Jr Front-end Web Developer",
        country: "Lebanon",
        id:8,
    },
]
// const members variable will be used as a placeholder for firebase. When we fetch there, this variable will be deleted

const COUNTRIES=["View All", "USA", "Turkey", "Lebanon"]

function Team() {
    const [category, setCategory]=useState("View All");


    const renderBtn = COUNTRIES.map(country=>{
        const className = country === category ? "rounded-full inline-block px-3 py-2 border-solid border-2 border-cyan-700 bg-cyan-700 text-white font-bold text-xl" : "rounded-full px-3 py-2 mx-1 text-cyan-700 text-xl font-bold";
        return (
            <div key={country} className="p-3">
          <button 
            className={className}
            onClick={()=> setCategory(country)}
          >{country}</button>
          </div>
        )}
      )

      const memberFilter = MEMBERS.filter((member)=> category==="View All" || category===member.country)

      const memberCard = memberFilter.map((member)=>{
        return(
            <div key={member.id} >
                <h3 >{member.name}</h3>
                <p >{member.occupation}</p>
            </div>
        )
      })

    return (
        <section name="team" >
            <h2 >Our Team</h2>
            <div name="country-filter" >
                {renderBtn}
            </div>
            <div name="team" >
                {memberCard}
            </div>
        </section>
    )
}

export default Team;