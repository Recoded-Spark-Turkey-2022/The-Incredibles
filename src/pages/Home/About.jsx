import React from 'react';
import Button from '../../components/Button';
import AboutHero from '../../assets/pics/homepage/about-hero.svg';
import AboutStar from '../../assets/pics/homepage/about-star.svg';

function About() {
  return (
    <section name="about" className="bg-emerald-50">
      <div
        name="about-header"
        className="text-center lg:p-20 lg:mx-10 max-lg:pt-10 "
      >
        <h1 className="text-cyan-600 lg:pb-8 font-bold text-4xl max-lg:mb-4">
          About
        </h1>

        <p className="mx-20 lg:px-40 lg:pb-10">
        Welcome to our website! We are a group of volunteers dedicated to helping refugees navigate their new lives and communities. 
        We understand that the journey to seeking refuge can be incredibly difficult and disorienting, and we want to do everything 
        we can to make the transition as smooth as possible.Our team is made up of individuals from all walks of life, united in our 
        belief that everyone deserves a chance at a better future. We offer a range of services, including language classes, 
        cultural orientation, legal assistance, and support finding housing and employment.We believe in the power of community and 
        connection, and we strive to create a welcoming and supportive environment for all of our members. 
        Thank you for visiting our website and for considering how you can help make a difference in the lives of refugees.

        </p>
        <div className="my-9">
          <Button name="Learn More" path="about" />
        </div>
      </div>
      <div
        name="about-body"
        className="flex lg:justify-between max-lg:flex-col-reverse max-lg:items-center max-lg:text-center lg:px-20"
      >
        <div name="cards" className="flex lg:justify-start ">
          <div name="left-side-cards" className="lg:">
            <div className="lg:p-6  max-w-sm mx-auto lg:space-x-4 ">
              <h2 className="flex  lg:justify-start lg:font-bold max-lg:flex-col max-lg:items-center">
                <img
                  src={AboutStar}
                  alt="about-star"
                  className="lg:mx-3 max-lg:w-10"
                />
                Dependability 
              </h2>
              <p className="lg:pt-6 flex-wrap">
              We are committed to being dependable and trustworthy, 
              and we work hard to follow through on our commitments 
              to refugees by giving fresh and truth information.{' '}
              </p>
            </div>
            <div className="lg:p-6 max-w-sm mx-auto space-x-4 max-lg:hidden ">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Determination
              </h2>
              <br />
              <p>
              We are determined to help refugees overcome obstacles and achieve their goals.
              </p>
            </div>
          </div>
          <div name="right-side-cards" className="max-lg:hidden">
            <div className="lg:p-6 max-w-sm mx-auto space-x-4 ">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Collaboration
              </h2>
              <br />
              <p>
              We believe that collaboration is key to achieving success, 
              and we work with a network of partners and community organizations to support refugees.
              </p>
            </div>
            <div className="lg:p-6 max-w-sm mx-auto space-x-4">
              <h2 className="flex justify-start font-bold">
                <img src={AboutStar} alt="about-star" className="mx-3" />
                Productivity
              </h2>
              <br />
              <p>
              We work efficiently and effectively to help refugees make the most of their opportunities.
              </p>
            </div>
          </div>
        </div>
        <div
          name="side-hero"
          className="flex justify-end max-lg:justify-center "
        >
          <img src={AboutHero} alt="about-hero" className="max-lg:w-1/2" />
        </div>
      </div>

      <div className="carousel-indicators right-0 bottom-0 left-0 hidden max-lg:flex justify-start lg:px-20 pb-20 m-10 max-lg:justify-center">
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="0"
          className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-cyan-600 bg-cyan-600"
          aria-current="true"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="1"
          className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="2"
          className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
        <button
          data-bs-target="#carouselDarkVariant"
          data-bs-slide-to="3"
          className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
          aria-current="false"
          aria-label="Slide 1"
        ></button>
      </div>
    </section>
  );
}

export default About;
