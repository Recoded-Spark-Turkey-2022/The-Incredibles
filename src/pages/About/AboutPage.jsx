import React from 'react';
import Story from './Story';
import CallToAction from './CallToAction';

// This component will be used as a container for About Page Components

function AboutPage() {
  return (
    <div name="aboutpage">
      <Story />
      {/* this part is a placeholder for Team component  */}
      {/* this part is a placeholder for Testimonials */}
      {/* this part is a placeholder for Partners */}
    <CallToAction/>
    </div>
  );
}

export default AboutPage;
