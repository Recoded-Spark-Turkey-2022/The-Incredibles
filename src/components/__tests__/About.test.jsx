import React from "react";
import renderer from "react-test-renderer";
import About from "../About";

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create(<About />).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });