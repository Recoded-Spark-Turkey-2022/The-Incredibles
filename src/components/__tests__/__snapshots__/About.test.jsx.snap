// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<section
  className="bg-emerald-50"
  name="about"
>
  <div
    className="text-center p-20 mx-10 ..."
    name="about-header"
  >
    <h1
      className="text-cyan-600 font-bold text-4xl"
    >
      About
    </h1>
    <br />
    <p
      className="mx-20"
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nulla est non repellat officia odio. Ipsam veniam, laudantium dolores praesentium nisi soluta impedit quae, suscipit fugit eveniet, qui tempora fugiat veritatis! Lorem ipsum dolor sit amet consectetur adipisicing elit. Vitae nostrum, sint eius voluptatum cum ratione labore expedita similique iste facilis sed totam cupiditate vero inventore reiciendis dolorem dolorum accusantium voluptatibus!
    </p>
    <br />
    <button
      className="inline-block px-10 py-2.5 bg-cyan-600 text-white font-medium text-l leading-tight  rounded-full shadow-md ease-in duration-300 hover:bg-purple-700 hover:shadow-lg my-9 hover:scale-110"
      type="button"
    >
      Learn More
    </button>
  </div>
  <div
    className="flex justify-between"
    name="about-body"
  >
    <div
      className="flex justify-start"
      name="cards"
    >
      <div
        name="left-side-cards"
      >
        <div
          className="p-6 max-w-sm mx-auto space-x-4"
        >
          <h2
            className="flex justify-start font-bold"
          >
            <img
              alt="about-star"
              className="mx-3"
              src="about-star.svg"
            />
            Reliability
          </h2>
          <br />
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque cum debitis repellendus placeat maiores eaque corrupti rerum? Reiciendis nisi voluptas corrupti, aspernatur autem est vero necessitatibus optio laudantium rerum numquam.
             
          </p>
        </div>
        <div
          className="p-6 max-w-sm mx-auto space-x-4"
        >
          <h2
            className="flex justify-start font-bold"
          >
            <img
              alt="about-star"
              className="mx-3"
              src="about-star.svg"
            />
            Efficiency
          </h2>
          <br />
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam, quidem sit minus dolore explicabo velit quas temporibus sequi vitae aut ducimus nostrum beatae eveniet atque ullam, quisquam aliquam sed est.
          </p>
        </div>
      </div>
      <div
        name="right-side-cards"
      >
        <div
          className="p-6 max-w-sm mx-auto space-x-4"
        >
          <h2
            className="flex justify-start font-bold"
          >
            <img
              alt="about-star"
              className="mx-3"
              src="about-star.svg"
            />
            Motivation
          </h2>
          <br />
          <p>
            Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptates, quaerat, molestiae vitae sed aliquid in nostrum laborum obcaecati optio expedita atque ab ipsam veritatis iure iusto similique quibusdam laboriosam sunt.
          </p>
        </div>
        <div
          className="p-6 max-w-sm mx-auto space-x-4"
        >
          <h2
            className="flex justify-start font-bold"
          >
            <img
              alt="about-star"
              className="mx-3"
              src="about-star.svg"
            />
            Creativity
          </h2>
          <br />
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Suscipit ex et commodi consequuntur, vel quis alias perferendis magni minima labore. Totam doloribus incidunt velit odio corrupti beatae quasi minima laudantium.
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex justify-end"
      name="side-hero"
    >
      <img
        alt="about-hero"
        src="about-hero.svg"
      />
    </div>
  </div>
  <div
    className="carousel-indicators right-0 bottom-0 left-0 flex justify-start pb-20 m-10"
  >
    <button
      aria-current="true"
      aria-label="Slide 1"
      className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-cyan-600 bg-cyan-600"
      data-bs-slide-to="0"
      data-bs-target="#carouselDarkVariant"
    />
    <button
      aria-current="false"
      aria-label="Slide 1"
      className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-gray-300"
      data-bs-slide-to="1"
      data-bs-target="#carouselDarkVariant"
    />
    <button
      aria-current="false"
      aria-label="Slide 1"
      className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
      data-bs-slide-to="2"
      data-bs-target="#carouselDarkVariant"
    />
    <button
      aria-current="false"
      aria-label="Slide 1"
      className="rounded-full p-1 mx-1 border-solid border-2 border-gray-300"
      data-bs-slide-to="3"
      data-bs-target="#carouselDarkVariant"
    />
  </div>
</section>
`;
